modules/promotion/commerce_promotion.module

```diff
@@ -79,3 +79,30 @@ function commerce_promotion_entity_base_field_info(EntityTypeInterface $entity_t
     return $fields;
   }
 }
+
+/**
+ * Implements hook_cron()
+ */
+function commerce_promotion_cron() {
+  commerce_promotion_disabled_expired();
+}
+
+/**
+ * Get all expired promotions that are still active and disable them.
+ */
+function commerce_promotion_disabled_expired() {
```

Comment:
If the storage provides a method to load expired, we can just keep this within the cron, no need for its own function. If people want to run this manually then they can.